src/ds++/DracoStrings.hh

```diff
@@ -262,6 +262,39 @@ inline std::string remove_color(std::string const &colored_string) {
   return std::regex_replace(colored_string, color_regex, "");
 }
 
+//----------------------------------------------------------------------------//
+/*!
+ * \brief Extract version of the form M.m.p from a string.
+ *
+ * \param[in] string_in String that contains a version number as a substring.
+ * \param[in] digits number of version digits to print. Default = 3.
+ * \return The version as a substring.
+ *
+ * Given the string "/ccs/opt/vendors-ec/ndi/2.1.3/share/gendir.all", return
+ * "2.1.3".
+ */
+inline std::string extract_version(std::string const &string_in,
```

Comment:
My understanding (I can't read regex though) is that this won't work for alpha/beta NDI deployments e.g. `2.1.3alpha/`. Maybe it's not worth agonizing about mismatches in those cases since they're alphas/betas, but just bringing it up now. An alternative would be to take a string as input, check that it is of the form "X.Y.Z", and then return three strings "X", "Y", "Z" to be compared with `std::string`'s `==` -- does that seem worthwhile to implement? If you think so I can make the modification.

Reply:
I forgot about the `alpha/beta` versions.  Let me extend the regex to account for this case.  Thank you for the thoughtful review.

Reply:
I also need to add a unit test in `ds++` for this new function.

Reply:
Also fixed.